cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -403,21 +403,45 @@ func ValidateIPNetFromString(subnetStr string, minAddrs int64, isDualStack bool,
 	return allErrs
 }
 
+// ValidatePodSubnetNodeMask validates that the relation between podSubnet and node-masks are correct
+func ValidatePodSubnetNodeMask(subnetStr string, c *kubeadm.ClusterConfiguration, isDualStack bool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// subnets were already validated
+	subnets, _ := utilnet.ParseCIDRs(strings.Split(subnetStr, ","))
+	for _, podSubnet := range subnets {
+		// obtain podSubnet mask
+		mask := podSubnet.Mask
+		maskSize, _ := mask.Size()
+		// obtain node-cidr-mask
+		nodeMask := componentconfigs.GetClusterNodeMask(c, utilnet.IsIPv6(podSubnet.IP), isDualStack)
+		// the pod subnet mask needs to allow one or multiple node-masks
+		// i.e. if it has a /24 the node mask must be between 24 and 32 for ipv4
+		if maskSize > nodeMask {
+			allErrs = append(allErrs, field.Invalid(fldPath, podSubnet.String(), "pod subnet mask is smaller than the node-mask"))
+		} else if (nodeMask - maskSize) > constants.PodSubnetNodeMaskMaxDiff {
+			allErrs = append(allErrs, field.Invalid(fldPath, podSubnet.String(), "pod subnet mask and node-mask different is greater than 16"))
+		}
+	}
+	return allErrs
+}
+
 // ValidateNetworking validates networking configuration
-func ValidateNetworking(c *kubeadm.ClusterConfiguration, fldPath *field.Path) field.ErrorList {
+func ValidateNetworking(cfg *kubeadm.ClusterConfiguration, fldPath *field.Path) field.ErrorList {
```

Comment:
is an unrelated change but using the  `c`  only was problematic to me during the refactor, I think that this  reads better

Reply:
it is not consistent with the rest of the functions though, was there an error if `c` was used for this function?

Reply:
Nope, will change it again, :)